### ISSUE TITLE
Validation fails when start date is 0001-01-01

### ISSUE THREAD

**Issue by snoopie72** [ewoutkramer/fhir-net-api#474]
We are trying to validate MedicationStatement, it fails with 'Instance failed constraint per-1 "If present, start SHALL have a lower value than end"'

Our input is this: 
```
<effectivePeriod>
<start value="0001-01-01" />
<end value="2016-04-07" />
</effectivePeriod>
```

When we change start value to '0001-01-02' it works fine.

**Comment by ewoutkramer** [comment]
I have found the cause: since you are (just like me) in GMT+1 (or +2 depending on DST), and the .NET API uses GMT-based DateTimeOffset internally, the actual date 0001-01-01 is parsed as being at +1 GMT, correcting this to GMT actually turns this date into 23:00 december 31st of the previous day, which is before 01-01-01 and thus unrepresentable in .NET.  This should actually throw an exception somewhere - this is probably caught and simply failure of per-1 is reported.

I do represent partial date times internally as a string (there's no native type to support it), but use PartialDateTime for comparisons.

The fix is simple, the parse of dates without a date-time should assume midnight GMT - not midnight local time.

**Comment by brianpos** [comment]
A wonder I never found this at +10

**Comment by ewoutkramer** [comment]
Fixed in #484
